lib/Partitioner/PartitionerUtils.cpp

```diff
@@ -41,24 +46,25 @@ BFSLevel getBFSLevel(Function *F) {
   BFSLevel bfs;
   int level = 0;
   int current = 0;
-  bfs.push_back({level, std::vector<Node *>()});
+  bfs.push_back(std::vector<Node *>());
   level++;
   while (current < level) {
     std::vector<Node *> nodes;
+    std::set<Node *> used;
```

Comment:
So `used` is there to make sure that the same node isn't inserted twice into `nodes`, right? Would it be possible to just move `nodes` and `cur` to be `std::unordered_set`, and then not introduce `used` at all? No big deal but maybe would make it a little simpler.